src/v/kafka/client/config_utils.cc

```diff
@@ -84,4 +85,14 @@ ss::future<> set_client_credentials(
     });
 }
 
+model::compression compression_from_str(std::string_view v) {
+    return string_switch<model::compression>(v)
+      .match("none", model::compression::none)
+      .match("gzip", model::compression::gzip)
+      .match("snappy", model::compression::snappy)
+      .match("lz4", model::compression::lz4)
+      .match("zstd", model::compression::zstd)
+      .default_match(model::compression::producer);
```

Comment:
> should the default here be `none`?  Isn't `producer` more of a broker option and not for the client?

I may not have enough context on the GitHub UI on mobile, but as far as I can tell, the topic-kevel option for compression is merely advisable.

1. It can change over time, which is why it's instilled into every batch header.
2. If compression results in a larger payload, it can be sent without compression (see franz-go).
3. Redpanda never recompresses regardless of the compression set on the topic or record batch

Reply:
Right, I understand that's the purpose at the topic level.  But this is for the config for kafka client

Reply:
Yep, I missed that. I agree with none.